src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -22,6 +23,119 @@
 OPENINFERENCE_QUERY_COLUMN_NAME = "attributes." + INPUT_VALUE
 OPENINFERENCE_DOCUMENT_COLUMN_NAME = "attributes." + RETRIEVAL_DOCUMENTS
 
+# argument keys in the default openai function call,
+# defined here only to prevent typos
+_RESPONSE = "response"
+_EXPLANATION = "explanation"
+
+
+class ClassificationResult(NamedTuple):
+    """
+    A tuple of classification label and (optional) explanation.
+    """
+
+    # Note that `NamedTuple` is as subclass of `tuple`, to allow for a
+    # seamless conversion to pandas dataframe, as in e.g.,
+    # `df = pd.DataFrame([ClassificationResult("relevant", None)])`
+    # or
+    # `df[["label", "explanation"]] = [ClassificationResult('label', None)]`
+    label: str
+    explanation: Optional[str] = None
+
+
+def llm_classify_with_explanation(
```

Comment:
I like this - though now that we have a "box" called ClassificationResult, I do feel it would be nicer to keep the API signature simpler long run. I wonder if we can figure out a migration strategy so we don't have to maintain a **_with_explanation for each variant.

Reply:
I think you're referring to ditching the `str` variant in favor of only the boxed version. Then that would be product question.

Reply:
Oh I didn't quite follow that "overload" didn't work. Is there no amount of ignores to support that?

In general I think I'd be pretty happy to ditch the `str` for the sake of keeping the footprint simpler and now having a unified return signature going forward. We can announce the breaking change. I think if we are going to make the change it would be now.

I think one signature would provide better ergonomics (not just reduce duplicate code paths) - the benefit being you can just "turn on" explanations when you need it. That definitely provides better ergonomics long run if you are iterating on evals (run with explanations for a sample, then turn it off when you increase the sweep)

Feel free to get feedback from others though.

Reply:
That makes sense to me. Let's chat about it next Monday when Xander is back and we can make a decision then.